### DISCUSSION TITLE
Getting FATAL no bootmgr.exe error

### DISCUSSION THREAD

**Discussion by chavan-dinesh** [ipxe/ipxe#1267] [Q&A]
I am getting Error while trying to PXE boot Oracle Virtual Box VM.
I tried downgraded versions of wimboot too but no luck!
Also, tried to pass initrd to mention bootmgr.exe but it throws an error unrecognized argument initrd.
Please help me it's really important to boot this windows VM via network.

![image](https://github.com/user-attachments/assets/f469fd73-c7b5-4353-8552-8363f8b2ecfc)

**Comment by NiKiZe** [comment]
Are you using iPXE or something else?
If `bootmgr.exe` isn't included in your `.wim` file, then you need to provide it thru initrd, just as the .wim is.

The best however, is to have a not to optimized wim that actually contains all files needed.

**Comment by chavan-dinesh** [comment]
@NiKiZe Thank you for your quick response! Really appreciate it Sir! bootmgr.exe does exist in path Windows/Boot/PXE in boot.wim.
I verified by mounting boot.wim on Windows machine. I tried booting windows VM using latest wimboot as well as older versions but it's throwing same error mentioned above. Also , how do we pass boot.wim file to PXE ?

**Comment by chavan-dinesh** [comment]
This is how my server side config file looks like:
```
root@ubuntu-test:/srv/tftp/pxelinux.cfg# cat default
DEFAULT menu.c32
PROMPT 0
MENU TITLE PXE Boot Menu
TIMEOUT 300

LABEL Install Windows
    MENU LABEL Install Windows
    KERNEL wimboot
```

**Reply by NiKiZe** [reply]
My bad, you do not have boot.wim loaded at all. (You should be able to verify this in your TFTP logs.) Not sure how to do that in pxelinux, might be as simple as initrd=boot.wim But since you are limited to TFTP, consider switching to iPXE and use http instead.
